packages/venia-ui/lib/components/ProductFullDetail/productFullDetail.js

```diff
@@ -126,10 +127,18 @@ const ProductFullDetail = props => {
     }
 
     const cartActionContent = isSupportedProductType ? (
-        <Button disabled={isAddToCartDisabled} priority="high" type="submit">
+        <Button
+            disabled={isAddToCartDisabled || isOutOfStock}
+            priority="high"
+            type="submit"
+        >
             <FormattedMessage
-                id={'productFullDetail.cartAction'}
-                defaultMessage={'Add to Cart'}
+                id={
```

Comment:
I would have preferred to have 2 different elements instead of the if-else in a single element for clarity (not that this isn't). What do you think about creating different elements for out of stock and add to cart and render one or the other depending on the situation? 

FYI, its a not a deal breaker, I am ok with this approach as well 😃

Reply:
Updated to have 2 elements